Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -36,16 +38,14 @@ public final class Chat {
         client: ChatClient,
         environment: Environment = .init()
     ) {
-        attachmentQueueUploader = client.attachmentQueueUploader
```

Comment:
This is accessing background workers in `ChatClient`. These are setup later when connecting a user, so I figured it is better to access these later in `Chat` as well but this requires that I store the `ChatClient` instance. Controllers do the same and I guess we need to have this shortcut available as a public API anyway.
From the design point of view it feels a little bit weird that a child object keeps a strong reference to the creator object but I guess it is fine.